metrics/bigquery.py

```diff
@@ -50,67 +50,159 @@ def validate_metric_name(name):
+        do_jq(config['jqfilter'], data_filename, filtered)
+
+        self.copy(filtered, os.path.join(config['metric'], filtered))
+        self.copy(filtered, latest)
+
+    def influx_upload(self, config, data_filename):
+        """Uses jq to extract InfluxDB time series points then uploads to DB."""
+        points = '%s-data-points.json' % config['metric']
+        jq_point = config.get('measurements', {}).get('jq', None)
+        if jq_point:
+            if self.influx:
+                do_jq(jq_point, data_filename, points)
+                with open(points) as points_file:
+                    try:
+                        points = json.load(points_file)
+                    except ValueError:
+                        print >>sys.stderr, "No influxdb points to upload.\n"
+                        return
+                points = [ints_to_floats(point) for point in points]
+                self.influx.write_points(points, time_precision='s')
+            else:
+                print >>sys.stderr, (
+                    'Skipping influxdb upload of metric %s, no db configured.\n'
+                    % config['metric']
                 )
-            influx.push(points, "metrics")
-        else:
-            print >>sys.stderr, 'Skipping upload of %s, no influxdb config.\n' % metric
-
-    def copy(src, dst):
-        """Use gsutil to copy src to test with minimal caching."""
-        check([
-            'gsutil', '-h', 'Cache-Control:max-age=60', 'cp', src, dst])
 
-    copy(raw, os.path.join(prefix, metric, raw))
-    copy(filtered, os.path.join(prefix, metric, filtered))
-    copy(filtered, os.path.join(prefix, latest))
+    def run_metric(self, config):
+        """Runs query and filters results, uploading data to GCS."""
+        raw = 'raw-%s.json' % time.strftime('%Y-%m-%d')
+
+        self.update_query(config)
+        self.do_query(config['query'], raw)
+        self.copy(raw, os.path.join(self.prefix, config['metric'], raw))
+
+        consumer_error = False
+        for consumer in self.result_consumers:
+            try:
+                consumer(config, raw)
+            except Exception: # pylint: disable=broad-except
+                print >>sys.stderr, traceback.format_exc()
+                consumer_error = True
+        if consumer_error:
+            raise ValueError('Error(s) were thrown by query result consumers.')
+
+    def copy(self, src, dest):
+        """Use gsutil to copy src to <bucket_path>/dest with minimal caching."""
+        dest = os.path.join(self.prefix, dest)
+        check(['gsutil', '-h', 'Cache-Control:max-age=60', 'cp', src, dest])
+
+    def update_query(self, config):
+        """Modifies config['query'] based on the metric configuration."""
+
+        # Currently the only modification that is supported is injecting the
+        # timestamp of the most recent influxdb data for a given metric.
+        # (For backfilling)
+        if (self.influx and
```

Comment:
Invert to increase indentation.

```
if not self.influx: return
if 'measurements' not in config: return
etc
```

Reply:
Done, but pylint made forced me to do some creative indentation. Is theres a cleaner format that pylint accepts?